src/main/kotlin/org/radarbase/push/integration/garmin/backfill/route/GarminRoute.kt

```diff
@@ -12,31 +21,43 @@ import java.time.Instant
 
 abstract class GarminRoute(
     private val consumerKey: String,
-    private val consumerSecret: String,
     private val userRepository: GarminUserRepository
 ) : Route {
     override val maxDaysPerRequest: Int
         get() = 5
 
-    fun createRequest(user: User, url: String): Request {
-
-        val oauth1 = Oauth1Signing(
-            oauthKeys = OauthKeys(
-                consumerKey,
-                consumerSecret,
-                userRepository.getAccessToken(user),
-                userRepository.getUserAccessTokenSecret(user)
-            )
-        )
-
+    fun createRequest(user: User, baseUrl: String, queryParams: String): Request {
         val request = Request.Builder()
-            .url(url)
+            .url(baseUrl + queryParams)
             .get()
             .build()
 
+        val accessToken = userRepository.getAccessToken(user)
+        val parameters = getParams(request.url)
+        val oauth1 = Oauth1Signing(parameters)
+
+        val signature = userRepository.getOAuthSignature(user, baseUrl, ROUTE_METHOD, parameters)
+        parameters[OAUTH_TOKEN] = accessToken
+        parameters[OAUTH_SIGNATURE] = signature.signedUrl
```

Comment:
Ah I was not clear about my intention with returning a `signedUrl`: my thought was that the full signed URL is all you need to make the eventual request. Since the signature depends on the ordering of the parameters, you will have a signature that is ordered identically to how the signature happened.

request

```json
{
    "url": "https://....?param1=value1"
}
```
and response
```json
{
    "url": "https://....?param1=value1",
    "signedUrl": "https://....?param1=value1&oauth_signature=abcdef"
}
```

Otherwise, it would be better to call the field `oauthSignature`. Then it might also make more sense to add the parameters as separate entities, e.g.
```json
{
   "url": "https:/....",
   "parameters": {
      "param1": "value1"
   }
}
```
and response
```json
{
   "url": "https:/....",
   "parameters": {
      "param1": "value1",
      "oauthSignature": "abcdef..."
   },
}
```
where the parameters are stored in a `LinkedHashMap` to preserve the ordering.

Reply:
Oh okay I see. Yes I am currently adding the parameters as separate entitites for the request. I will change the response as well. But I'm currently not sending the access token but inserting them on both sides separately (here in the Push Endpoint and before signing in the Rest Source Auth). Should I just pass it as a parameter as well?

Reply:
In any case, I think the response should contain all information to actually make the request, including the access token. The request could omit the access token and oauth signature type.

Reply:
Okay I see thanks

Reply:
I've made the changes. I've also made the changes here https://github.com/RADAR-base/RADAR-Rest-Source-Auth/pull/97